runtime/shibuya/src/lib.rs

```diff
@@ -229,6 +230,19 @@ impl_opaque_keys! {
     }
 }
 
+// Adjusting the base extrinsic weight to account for the additional database
+// read introduced by the `tx-pause` pallet during extrinsic filtering.
+//
+// TODO: This hardcoded addition is a temporary fix. Replace it with a proper
+// benchmark in the future.
+pub struct RuntimeAdjustedWeights;
+impl RuntimeAdjustedWeights {
+    pub fn extrinsic_base_weight() -> Weight {
+        ExtrinsicBaseWeight::get()
+            .saturating_add(<Runtime as frame_system::Config>::DbWeight::get().reads(1))
+    }
+}
```

Comment:
I'd rather suggest to define something like `AdjustedBaseWeight` and implement `Get` on top of it.

Minor comment though.

Reply:
Yes, @Dinonard is right, you can use
```rust
fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
        let p = WeightFeeFactor::get();
        let q = Balance::from(RuntimeBlockWeights::get().get(DispatchClass::Normal).base_extrinsic.ref_time());
```